content/proof-resource/index.md

```diff
@@ -0,0 +1,174 @@
+---
+rfc: 0023
+start_date: 2018-10-01
+decision_date:
+pr: openregister/registers-rfcs/pull/40
+status: draft
+---
+
+# Proof resource
+
+## Summary
+
+This RFC proposes to fix the type of proofs to Certificate Transparency's
+Merkle tree approach.
+
+## Motivation
+
+Currently, the specification define the Proof resource endpoints parametrized
+by type of proof through the `proof-identifier` parameter. The implications of
+implementing a proof system are substantial as demonstrated by the long time
+is taking the reference implementation to have a stable implementation for a
+SHA2-256 Merkle tree. Also, a Register uses a single hashing function which
+means the proof type has to align with that fact.
+
+## Explanation
+
+The proposal is to restrict the possible proofs to Merkle-based (i.e.
+Certificate Transparency based) ones. And rely on the register hashing
+function to determine what to use to compute the root hash.
+
+### Endpoints
+
+The implications for each endpoint are mainly the removal of the paramter
+`proof-identifier` and the pluralisation of the resource scope to align with
+the rest of the API.
+
+#### Register proof
+
+```
+GET /proofs/register
+```
+
+##### Response attributes
+
+|Name|Type|Description|
+|-|-|-|
+|`root-hash`|Hash|The root hash for the log when the proof was issued.|
+|`total-entries`|Integer|The size of the log when the proof was issued.|
+
+Note that `tree-head-signature` will be addressed in another RFC when we have
+done the work to scope implications.
+
+***
+**EXAMPLE:**
+
+For example, a register using a SHA2-256 hashing function:
+
+```http
+GET /proofs/register HTTP/1.1
+Host: country.register.gov.uk
+Accept: application/json
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "root-hash": "12208d92e1e0af1d43c41e498e6baed0d0b3ea2770d1bf9d2afc04e9c4dad7795729",
+  "total-entries": 208
+}
+```
+***
+
+#### Consistency proof
+
+```
+GET /proofs/consistency/{small-log-size}/{large-log-size}
```

Comment:
Am I right in thinking that the most common use case will be for `large-log-size` to be the current size of the log? If so, can we make it optional?

Reply:
I think it is indeed the most common use case. Do you think we need to address (agree?) on the optional bit in this RFC or are you comfortable to accept it as it is now, and address the implications for an optional bit in another RFC?

Reply:
Either way is fine by me 👍

Reply:
Ook, I'll merge then 👍